### PR TITLE
Allocator support exitOnOutOfMemory config.

### PR DESCRIPTION
**Pull request by horizonzy**
Descriptions of the changes in this PR:
In some cases, if the process can't allocate memory, we would better exit the JVM, or the bookie can't serve, and the client always timeout.
Add config `allocatorExitOnOutOfMemory`. When the allocator encounters oom, exit the JVM.

### CODE REVIEW

**Comment by hangc0276** [comment]
@horizonzy Please fix the check style, thanks.

**Review by hangc0276 — APPROVED** [review]
Good job!

**Comment by hangc0276** [comment]
@lhotari Please help take a look, thanks.

**Comment by horizonzy** [comment]
Some test fails. I will submit a PR to fix the failed test.

**Comment by horizonzy** [comment]
> The ci failed. Fixing it now.

done

**Review by dlg99 — APPROVED** [review]
great work!

**Comment by horizonzy** [comment]
rerun failure checks

**Review by hezhangjian — CHANGES_REQUESTED** [review]
could you please rebase this code? we have deprecated powermock usage now.

**Review by StevenLuMT — APPROVED** [review]
LGTM

File: bookkeeper-common-allocator/src/main/java/org/apache/bookkeeper/common/util/ShutdownUtil.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Forked from <a href="https://github.com/apache/pulsar">Pulsar</a>.
+ */
+@Slf4j
+public class ShutdownUtil {
+    private static final Method log4j2ShutdownMethod;
```
[the remark this answers is not in the document]
Comment (horizonzy) [reply bookkeeper-common-allocator/src/main/java/org/apache/bookkeeper/common/util/ShutdownUtil.java:30]: It's a minor import, we needn't import new dependency.

**Review by hezhangjian — APPROVED** [review]

**Comment by StevenLuMT** [comment]
image
please fix the checkstyle issue, thanks @horizonzy

**Comment by hangc0276** [comment]
@StevenLuMT This PR changed the interface, and we should not cherry-pick it to the patch branch.

**Pull request merged**